### PR TITLE
Create templates for monthly reports

### PR DESCRIPTION
**Pull request by chambridge**
* Remove reports without start/end dates
* Add templates for generating reports

### CODE REVIEW

File: roles/setup/tasks/dates.yml [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+- name: Set report_start_month if single digit
+  set_fact:
+    report_start_month: "{{ '0' + report_start_month }}"
+  when:
+    - (current_month | int) <= 10
+
+- name: Set report_end_year to current year
+  set_fact:
+    report_end_year: "{{ (ansible_date_time.year }}"
```
Comment (abaiken) [inline roles/setup/tasks/dates.yml:39]:
```suggestion
    report_end_year: "{{ (ansible_date_time.year) }}"
```

Comment (abaiken) [reply roles/setup/tasks/dates.yml:39]: Changing the above ^ allowed tests to pass for me locally.

**Comment by chambridge** [comment]
Items left to manually test:

# Basic:
- [ ] Deployment works as expected and creates reports for the current month
- [ ] Current month reports work without error (can use start at beginning of month when the report is created in the middle of the month)

# Next Month Report Creation:
Deploy operator using `current_day`, `current_month`, `current_year` overrides to simulate the day of the month is 25 or later to trigger the creation of the next months report and various test for the month/year.

## Single digit month
- [ ] Verify reports are created for the next month and are formatted correctly (prefix of `0`) for the month value

## Double digit month
- [ ] Verify reports are created for the next month and are formatted correctly for the month value

## December month
- [ ] Verify reports are created for the next month (in the next year) and are formatted correctly for the month & year values

# Previous Month Report Removal:
Deploy operator using `current_day`, `current_month`, `current_year` overrides to simulate the day of the month is 3 or earlier to trigger the removal of the previous months report and various test for the month/year.

## Single digit month
Manually create reports for previous month (Only the report name must match contents are less important)
- [ ] Verify reports are deleted for the previous month

## Double digit month
Manually create reports for previous month (Only the report name must match contents are less important)
- [ ] Verify reports are deleted for the previous month

## January month
Manually create reports for previous month/year (Only the report name must match contents are less important)
- [ ] Verify reports are deleted for the previous month/year

**Comment by chambridge** [comment]
Have some more work here to do:
```
failed to resolve ReportQuery dependencies cm-openshift-usage-lookback: report.metering.openshift.io "cm-openshift-usage" not found
```
Lookback reports depend on report queries that build on reports, so those report queries need to templated also.

**Comment by chambridge** [comment]
The lookback report issue should be resolved now.

Testing may continue.

File: roles/collect/tasks/main.yml [hunk 2]
```diff
@@ -176,6 +192,11 @@
   set_fact:
     source_commit: "{{ lookup('file', ansible_env.HOME + '/commit') }}"
 
+- name: Obtain the source commit from file
+  set_fact:
+    source_commit: "{{ lookup('file', ansible_env.HOME + '/commit') }}"
+  when: source_commit is not defined or source_commit == ''
+
```
Comment (abaiken) [inline roles/collect/tasks/main.yml:199]:
```suggestion
```

Comment (chambridge) [reply roles/collect/tasks/main.yml:199]: huh ... bad merge I guess 😞

**Pull request merged**